### PR TITLE
Add missing variable width operations to standard_gate_name_mapping()

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

In #8852 we added support to the target class for representing variable width operations by using the class directly instead of an instance. Now that we have settled on a representation for these types of operations we can add them to the built in name mapping function (from which they were previously excluded. This commit adds the built-in variable width operations to the output of the function and returns just the class instead of an instance.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
I was originally thinking we should backport this, but since it's arguably an API change for the function (since it'll return a class for the variable width ops instead of an instance) so I don't think we should.

File: qiskit/circuit/library/standard_gates/__init__.py [hunk 1]
```diff
@@ -107,14 +107,20 @@
...
+
+    For variadic operations (i.e. operations that operate on a variable
+    number of qubits), such as :class:`~.MCXGate`, the class for that
+    operation is returned and not an instance.
+    """
     from qiskit.circuit.parameter import Parameter
     from qiskit.circuit.measure import Measure
     from qiskit.circuit.delay import Delay
     from qiskit.circuit.reset import Reset
+    from qiskit.circuit.controlflow import IfElseOp, WhileLoop, ForLoop
```
Comment (jakelishman) [inline qiskit/circuit/library/standard_gates/__init__.py:120]: I'm not really certain we should be returning these as "standard gates"; they're no gates, and they don't really follow a standard form of accepting n float-like parameters.

I suppose it depends a bit on how the result of this function is meant to be used.

File: qiskit/circuit/library/standard_gates/__init__.py [hunk 2]
```diff
@@ -171,4 +177,15 @@ def get_standard_gate_name_mapping():
         Measure(),
     ]
     name_mapping = {gate.name: gate for gate in gates}
+    # Add variable width gates as a mapping between names and classes
+    # since an instance requires knowing how many qubits are being operated on
+    name_mapping["if_else"] = IFElseOp
```
Comment (jakelishman) [inline qiskit/circuit/library/standard_gates/__init__.py:182]: typo here (extra capital letter)

File: qiskit/circuit/library/standard_gates/__init__.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/circuit/library/standard_gates/__init__.py:120]: Yeah, I was on the fence about this too, but we already had `Delay`, `Reset`, etc here. This was added in #8759 so we had a mapping for `basis_gate` entries to object for a target. I guess the function name isn't great and we should rename it (well, alias, then deprecate, then rename).

File: qiskit/circuit/library/standard_gates/__init__.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/circuit/library/standard_gates/__init__.py:182]: Sigh, yeah and I missed an op below

Comment (mtreinish) [reply qiskit/circuit/library/standard_gates/__init__.py]: Fixed in: https://github.com/Qiskit/qiskit-terra/pull/8935/commits/210553e8fbab53130255c8bcdf2db9f42b31c87a

**Pull request closed without merging**